### PR TITLE
Throttle: complete all m_blockers when we set average or max to 0

### PR DESCRIPTION
**Pull request by yangdongsheng**
When we change the average or max to 0, we need to complete all m_blockers
first.

Fixes: http://tracker.ceph.com/issues/36715
Signed-off-by: Dongsheng Yang <dongsheng.yang@easystack.cn>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/librbd/io/ImageRequestWQ.cc [hunk 1]
```diff
@@ -642,9 +642,9 @@ bool ImageRequestWQ<I>::needs_throttle(ImageDispatchSpec<I> *item) {
     throttle = t.second;
     tokens = item->tokens_requested(flag);
 
-    if (throttle->get<ImageRequestWQ<I>, ImageDispatchSpec<I>,
-	      &ImageRequestWQ<I>::handle_throttle_ready>(
-		tokens, this, item, flag)) {
+    if (tokens && throttle->get<ImageRequestWQ<I>, ImageDispatchSpec<I>,
+	            &ImageRequestWQ<I>::handle_throttle_ready>(
```
Comment (dillaman) [inline src/librbd/io/ImageRequestWQ.cc]: Nit: hard to follow the w/ this indentation. Perhaps leave ```tokens &&``` on its own line for clarity

File: src/librbd/io/ImageRequestWQ.cc [hunk 2]
```diff
@@ -642,9 +642,10 @@ bool ImageRequestWQ<I>::needs_throttle(ImageDispatchSpec<I> *item) {
     throttle = t.second;
     tokens = item->tokens_requested(flag);
 
-    if (throttle->get<ImageRequestWQ<I>, ImageDispatchSpec<I>,
-	      &ImageRequestWQ<I>::handle_throttle_ready>(
-		tokens, this, item, flag)) {
+    if (tokens &&
+          throttle->get<ImageRequestWQ<I>, ImageDispatchSpec<I>,
+	    &ImageRequestWQ<I>::handle_throttle_ready>(
+	      tokens, this, item, flag)) {
```
Comment (yangdongsheng) [inline src/librbd/io/ImageRequestWQ.cc]: As it looks different in github about the indentation. I pasted the original code here, each indentation is two spaces. 
```
    if (tokens &&
          throttle->get<ImageRequestWQ<I>, ImageDispatchSpec<I>,
            &ImageRequestWQ<I>::handle_throttle_ready>(
              tokens, this, item, flag)) {
      blocked = true;
    } else {
      item->set_throttled(flag);
    }
```

**Comment by yangdongsheng** [comment]
@dillaman  rebased.

**Comment by yangdongsheng** [comment]
@dillaman Please help to review this PR, thanx
